Replace "bug report" issue template with issue forms.

ScottTodd opened the pull request:
Progress on https://github.com/google/iree/issues/7890. See GitHub's docs on [creating issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms).

Try this out here: https://github.com/ScottTodd/iree/issues/new/choose. Here's an example issue: https://github.com/ScottTodd/iree/issues/6 (note the "_No response_", how the multiple choice selection uses commas, and how the sections are set up. Definitely open to trying more ways to make both the form and the final submitted issues more useful.

It would also be nice to link to a page dedicated to getting useful error reports, maybe with a flow chart or something to help direct users coming in from TOSA/MHLO/etc.

GMNGeoffrey reviewed (approved):
Nice! It would be great if we could set labels based on the components, but this is great

benvanik reviewed (changes requested):
Let's remove "ask for help with something" from the description - it'd be really nice to not have people asking questions in issues. A communications channel link is fine, but making it simple for people to find their way into the right place would be nice by directly linking to the discussions.

ScottTodd commented:
> Let's remove "ask for help with something" from the description - it'd be really nice to not have people asking questions in issues. A communications channel link is fine, but making it simple for people to find their way into the right place would be nice by directly linking to the discussions.

Good point. I was mostly trying to pad this out with more text: 
![image](https://user-images.githubusercontent.com/4010439/172269677-3f735f38-874b-48f2-91a9-29c9bb6c2e8d.png)

I'll think on that some and make some tweaks tomorrow.

benvanik commented:
Maybe an easy solution is make "user report" more specific - "Bug Report" implies there's a bug vs. a feature request or help with using something/etc. We can't do much if someone sees "Bug Report" or "Feature Request" templates and asks a question about something random, but at least we'd have tried :P

ScottTodd commented:
> Maybe an easy solution is make "user report" more specific - "Bug Report" implies there's a bug vs. a feature request or help with using something/etc. We can't do much if someone sees "Bug Report" or "Feature Request" templates and asks a question about something random, but at least we'd have tried :P

So looking through recent issues filed by users (i.e. not core project developers), I think most of our issues are "bug reports" (missing lowering paths, correctness issues, build failures). We have a few that are "discussions" (asking about a lowering path or backend/deployment configuration), and a few that are "feature requests" (add test coverage, support a new platform).

I could split this "user report" form back into a "bug report" form and a "feature request" form and specialize a bit
* bug report would include repro steps and tips about what to include
* feature request could reference communication channels like the mailing list and MLIR Discourse forum more directly

ScottTodd commented:
PTAL - added a config with links to other communication channels and split the user report form into a bug report form and a feature request form. Also added emojis :)

GMNGeoffrey commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -0,0 +1,11 @@
+    about: For topics that don't fit into a more specific category
+  - name: 📧 Send an email to our mailing list
+    url: https://groups.google.com/forum/#!forum/iree-discuss
+    about: For announcements and lower priority discussion
```
I don't think we should say that the email list is for lower priority discussions *and* announcements. That doesn't make much sense to me. AFAIK we also never really had  discussion about using GitHub discussions, so now we're kind of just fragmented between even more channels. We could of course switch (e.g. make the email list for announcements only), but I don't think a PR is the right place to make that switch. So I would put the email list and discussions on equal footing. I actually didn't know we were using discussions at all...

ScottTodd replied:
Yeah, agreed about iterating on this. I was referencing the existing wording at https://google.github.io/iree/#communication-channels and then trying to differentiate between GitHub Discussions and the mailing list. We could just list one or the other here. If someone is already on GitHub, maybe it's easier to just direct to the discussions instead of off to the mailing list.

ScottTodd replied:
I can't think of a better description, so I just made them both say "For topics that don't fit into a more specific category". Any suggestions?

GMNGeoffrey replied:
I don't really understand what the "lower priority" was referring to regarding the mailing list. Where do "higher priority" discussions go? I think it's meant to be contrast with realtime discussion on Discord?

How about listing email list first and then GitHub discussion as a new alternative with similar scope? I'll attach a suggestion

GMNGeoffrey commented on .github/ISSUE_TEMPLATE/feature_request.yml:
```diff
@@ -0,0 +1,38 @@
+name: ➕ Feature Request
+description: Ask for a new feature to be added
+labels: ["awaiting-triage", "enhancement ➕", "help wanted"]
```
Can we just drop the "help wanted" label? I have no idea what it indicates

benvanik commented:
whoa this is amazing - I had no idea you could add the link callouts!
![image](https://user-images.githubusercontent.com/75337/172501331-625acf79-4246-4ca8-83aa-dc826dade513.png)

GMNGeoffrey commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -0,0 +1,11 @@
+blank_issues_enabled: true
+contact_links:
+  - name: 🗣 Start a discussion on GitHub
+    url: https://github.com/google/iree/discussions
+    about: For topics that don't fit into a more specific category
+  - name: 📧 Send an email to our mailing list
+    url: https://groups.google.com/forum/#!forum/iree-discuss
+    about: For topics that don't fit into a more specific category
+  - name: 💬 Join us on Discord
+    url: https://discord.gg/26P4xW4
+    about: For realtime communication with the team and collaborators
```
```suggestion
blank_issues_enabled: true
contact_links:
  - name: 📧 Send an email to our mailing list
    url: https://groups.google.com/forum/#!forum/iree-discuss
    about: For announcements and asynchronous discussion
  - name: 🗣 Start a discussion on GitHub
    url: https://github.com/google/iree/discussions
    about: An alternative platform for asynchronous discussion
  - name: 💬 Join us on Discord
    url: https://discord.gg/26P4xW4
    about: For realtime communication with the team and collaborators
```

GMNGeoffrey reviewed (approved):
Ok enough bike shedding. This is a huge improvement, thank you :-)
